File: _scratch/run/minimum_add_to_make_parentheses_valid/sol.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function(s) {
    let open = 0, add = 0;
    for (const c of s) {
        if (c === '(') open++;
        else if (open > 0) open--;
        else add++;
    }
    return add + open;
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(minAddToMakeValid(_s)));}
console.log(_o.join('\n'));

File: _scratch/run/minimum_add_to_make_parentheses_valid/sol.py
class Solution:
    def minAddToMakeValid(self, s):
        open_ = add = 0
        for c in s:
            if c == "(":
                open_ += 1
            elif open_ > 0:
                open_ -= 1
            else:
                add += 1
        return add + open_
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().minAddToMakeValid(_s)))
print('\n'.join(_o))